backend-lib/src/main/rust/lib.rs

```diff
@@ -396,7 +396,7 @@ fn encode_usk<'a>(
 }
 
 fn decode_usk(env: &JNIEnv, usk: JByteArray) -> anyhow::Result<UnifiedSpendingKey> {
-    let usk_bytes = SecretVec::new(env.convert_byte_array(usk).unwrap());
+    let usk_bytes = parse_secret(env, &usk)?;
```

Comment:
Nit: `parse_secret` should be called `secret_from_jni` (like `account_id_from_jni`), because it isn't doing any parsing.